cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/connector/resource/AzureDatabaseResourceService.java

```diff
@@ -112,52 +116,83 @@
+        String resourceGroupName = azureResourceGroupMetadataProvider.getResourceGroupName(cloudContext, stack);
+        LOGGER.debug("Deleting resource group {}", resourceGroupName);
         try {
             client.deleteResourceGroup(resourceGroupName);
         } catch (CloudException e) {
-            String errorMessage = null;
-            if (e.body() != null) {
-                String errorCode = e.body().code();
-                if ("ResourceGroupNotFound".equals(errorCode)) {
-                    LOGGER.warn("Resource group {} does not exist, assuming that it has already been deleted", resourceGroupName);
-                    // leave errorMessage null => do not throw exception
-                } else {
-                    String details = e.body().details() != null ? e.body().details().stream().map(CloudError::message).collect(Collectors.joining(", ")) : "";
-                    errorMessage = String.format("Resource group %s deletion failed, status code %s, error message: %s, details: %s",
-                            resourceGroupName, errorCode, e.body().message(), details);
-                }
-            } else {
-                errorMessage = String.format("Resource group %s deletion failed: '%s', please go to Azure Portal for details",
-                        resourceGroupName, e.getMessage());
-            }
-
-            if (errorMessage != null) {
-                if (force) {
-                    LOGGER.warn(errorMessage);
-                    LOGGER.warn("Resource group {} deletion failed, continuing because termination is forced", resourceGroupName);
-                } else {
-                    throw new CloudConnectorException(errorMessage, e);
-                }
-            }
+            errorHandling(force, e, "resource group", resourceGroupName);
         }
 
         return Lists.newArrayList(new CloudResourceStatus(CloudResource.builder()
-                .type(ResourceType.AZURE_RESOURCE_GROUP)
+                .type(AZURE_RESOURCE_GROUP)
                 .name(resourceGroupName)
                 .build(), ResourceStatus.DELETED));
     }
 
+    private List<CloudResourceStatus> deleteDatabaseServer(List<CloudResource> resources, boolean force, AzureClient client) {
+        LOGGER.debug("Deleting database server");
+        Optional<CloudResource> dbServerResourceOptional = resources.stream().filter(r -> AZURE_DATABASE.equals(r.getType())).findFirst();
+        if (dbServerResourceOptional.isEmpty()) {
+            LOGGER.warn("Azure database id not found in database, deleting nothing");
+            return List.of();
+        }
+        String databaseServerId = dbServerResourceOptional.get().getReference();
+        try {
+            azureUtils.deleteDatabaseServer(client, databaseServerId);
+        } catch (CloudException e) {
+            errorHandling(force, e, "Database server", databaseServerId);
+        }
+
+        return Lists.newArrayList(new CloudResourceStatus(CloudResource.builder()
```

Comment:
We should consider deleting the actual resource as it won't be reused in future.